File: CSE241 - Object Oriented Programming/studies/java/interfaces/payable/Employee.java
package payable;

//! be careful Comparable<Employee>, otherwise it become Comparable<Object>
public abstract class Employee implements Payable, Comparable<Employee> {
    private String _fname;
    private String _lname;
    private String _ssn;

    public Employee (String fname, String lname, String ssn) {
        _fname = fname;
        _lname =lname;
        _ssn = ssn;
    }

    public void setFirstName (String name) {_fname = name;}

    public void setLastName (String name) {_lname = name;}

    public void setSsn (String ssn) {_ssn = ssn;}

    public String firstName () {return _fname;}

    public String lastName () {return _lname;}

    public String ssn () {return _ssn;}

    public String toString () {
        return String.format("%s %s \nsocial security number: %s", firstName(), lastName(), ssn());
    }

    public int compareTo (Employee other) {
        if (paymentAmount() > other.paymentAmount())
            return 1;
        else if (paymentAmount() < other.paymentAmount())
            return -1;
        else 
            return 0;
    }

    // Note: We do not implement Payable method paymentAmount here so 
    // this class must be declared abstract to avoid a compilation error.
}

File: CSE241 - Object Oriented Programming/studies/cpp/ch15-Polymorphism-and-Virtual-Functions/employee/employee.cpp
#include <iostream>
using std::cout;
#include "employee.h"

namespace GTUEmployee {
    Employee::Employee (const string & firstName, const string & lastName, const string ssn)
    : _firstName(firstName), _lastName(lastName), _ssn(ssn) {}

    void Employee::setFirstName (const string & firstName) {_firstName = firstName;}
    void Employee::setLastName (const string & lastName) {_lastName = lastName;}
    void Employee::setSsn (const string & ssn) {_ssn = ssn;}

    string Employee::getFirstName () const {return _firstName;}
    string Employee::getLastName () const {return _lastName;}
    string Employee::getSsn () const {return _ssn;}

    void Employee::print() const {
        cout << getFirstName() << ' ' << getLastName()
             << "\nsocial security number: " << getSsn();
    }
}